src/main/java/com/zipcodewilmington/looplabs/IntegerDuplicateDeleter.java

```diff
@@ -1,8 +1,79 @@
 package com.zipcodewilmington.looplabs;
 
+import java.util.Arrays;
+
 /**
  * Created by leon on 1/29/18.
  * @ATTENTION_TO_STUDENTS You are forbidden from modifying the signature of this class.
  */
 public final class IntegerDuplicateDeleter extends DuplicateDeleter<Integer> {
-}
+
+    public IntegerDuplicateDeleter(Integer[] intArray) {
+        super(intArray);
+    }
+
+    @Override
+
+    public Integer[] removeDuplicates(int maxNumberOfDuplications) {
+        String nonDups = "";
+        int count = 0;
+
+        for (int i = 0; i < this.array.length; i++) {
+            for (int k = 0; k < this.array.length; k++) {
+                if (this.array[i].equals(this.array[k])) {
+                    count++;
+                }
+            }
+            if (count < maxNumberOfDuplications) {
+                nonDups += this.array[i] + " ";
+            }
+            count = 0;
+        }
+        if (nonDups == "") {
+            String[] emptyArray = new String[0];
```

Comment:
I don't understand what this is here for? If every item is a dupe, shouldn't you just return an empty array?